brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -83,7 +83,10 @@ public SpanId startNewSpan(String component, String operation) {
     }
 
     private SpanId getNewSpanId() {
-        Span parentSpan = spanAndEndpoint().state().getCurrentServerSpan().getSpan();
+        Span parentSpan = spanAndEndpoint().state().getCurrentLocalSpan();
```

Comment:
this is dodgy with the default state backend because people may not have cleared their local span and all of a sudden the traces start having random local spans as their parent. I'd make an internal flag that's set when the state store is InheritableServerClientAndLocalSpanState. Only when that's set accept a local parent.

While seems yakky, this will insure existing users against the behavior change I mentioned. It is super confusing to try and debug spans that have the wrong parent, speaking from experience!